lib/ex_unit/test/ex_unit/doc_test_test.exs

```diff
@@ -253,13 +253,15 @@ defmodule ExUnit.DocTestTest do
            test/ex_unit/doc_test_test.exs:117: ExUnit.DocTestTest.Invalid (module)
     """
 
+    # The stacktrace points to the cause of the error
     assert output =~ """
       4) test moduledoc at ExUnit.DocTestTest.Invalid (4) (ExUnit.DocTestTest.ActuallyCompiled)
          test/ex_unit/doc_test_test.exs:218
          Doctest failed: got UndefinedFunctionError with message undefined function: Hello.world/0 (module Hello is not available)
          code:  Hello.world
          stacktrace:
-           test/ex_unit/doc_test_test.exs:117: ExUnit.DocTestTest.Invalid (module)
+           Hello.world()
+           (for doctest at) test/ex_unit/doc_test_test.exs:117
```

Comment:
@josevalim that's close but does not render de same output, the line number is postponed after the module, not the file:
```
 (for doctest at) lib/string_calculator.ex (StringCalculator):52
```

What I've been able to do is replace the `"(for doctest at)"` frame with the synthetic one… a bit verbose:

```
merge = fn(original, synthetic) ->
    doctest_index = Enum.find_index original, fn(stacktrace_entry) ->
      location = elem(stacktrace_entry, tuple_size(stacktrace_entry) - 1)
      String.starts_with?(to_string(location[:file]), "(for doctest at)")
    end
    List.replace_at(original, doctest_index, List.last(synthetic))
  end
  merged_stack = merge.(original_stack, stack)
```

Note that I'm using an anonymous function. I tried using a `defp` but then it was not found (I'm guessing something related to macro expansion?).

I'm inclined to just using the raised stacktrace. There's some work done to annotate the source as a doctest that wouldn't be visible otherwise :-). Not sure what's the value of displaying the module?